### DISCUSSION TITLE
ReferenceError: Cypress/Cy is not defined

### DISCUSSION THREAD

**Discussion by johnsonkandul-github** [cypress-io/cypress#26130] [Questions and Help]
I have created new JS file called as email.js under cypress/support directory along side command.js, e2e.js, index.js files. 
Cypress version is latest 12.7.0.

I am trying to use cy.task(it is registered in cypress.config.js setupnodes events) or Cypress.config('baseURL') but getting below errors

Sample Errors :

    ReferenceError: Cypress is not defined
    List item TypeError: Cypress.config is not a function
    ReferenceError: Sendgrid_API_KEY is not defined

I have tried to import cypress using const   Cypress   = require("cypress"); however no luck. 
I am wondering why cypress() cy() works fines in js files under e2e and support folders without importing cypress

I have also tried to create new .eslintrc.json file and added below code but still no luck

{
  "plugins": [
    "cypress"
  ]
}

**Comment by portyu9** [comment]
Hi @johnsonkandul-github 

Putting a file under `cypress/support` does not automatically give it the Cypress browser globals.

`cy` and the browser-side `Cypress` object only exist when that code is actually bundled/executed by Cypress as test/support code.

This does **not** create that global:

```js
const Cypress = require('cypress')
```

`require('cypress')` gives you Cypress's Node package/API, not the `window.Cypress` object used inside a running test.

If your email code needs SendGrid and secrets, it is better implemented on the **Node side** and exposed to tests through `cy.task()`:

```js
// cypress.config.js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      on('task', {
        async sendEmail(payload) {
          await sendEmailWithSendGrid(
            payload,
            process.env.SENDGRID_API_KEY
          )

          return null
        },
      })
    },
  },
})
```

Then from a Cypress spec/support command:

```js
cy.task('sendEmail', payload)
```

A couple of other details from the code:

- The Cypress configuration key is `baseUrl`, not `baseURL`.
- `Sendgrid_API_KEY` should not be an undeclared variable. For Node-side secret access use something such as `process.env.SENDGRID_API_KEY`.
- Keep the SendGrid key out of browser-side code.

So first decide where `email.js` executes. If it is Node-side, do not use `cy` or browser `Cypress` there. If it needs to be triggered by a test, register the operation as a task.

Please mark this answer as accepted if it helped, thank you.
